pkg/agent/util/net_windows.go

```diff
@@ -14,7 +14,191 @@
+	LocalHNSNetwork     = "antrea-hnsnetwork"
+	OVSExtensionID      = "583CC151-73EC-4A6A-8B47-578297AD7623"
 )
+
+// EnableHostInterface sets the specified interface status as UP.
+func EnableHostInterface(ifaceName string) error {
+	cmd := fmt.Sprintf("Enable-NetAdapter -InterfaceAlias %s", ifaceName)
+	return invokePSCommand(cmd)
+}
+
+// ConfigureInterfaceAddress adds IPAddress on the specified interface.
+func ConfigureInterfaceAddress(ifaceName string, ipConfig net.IPNet) error {
+	ipStr := strings.Split(ipConfig.String(), "/")
+	cmd := fmt.Sprintf("New-NetIPAddress -InterfaceAlias %s -IPAddress %s -PrefixLength %s", ifaceName, ipStr[0], ipStr[1])
+	return invokePSCommand(cmd)
+}
+
+// EnableIPForwarding enables the IP interface to forward packets that arrive on this interface to other interfaces.
+func EnableIPForwarding(ifaceName string) error {
+	cmd := fmt.Sprintf("Set-NetIPInterface -InterfaceAlias %s -Forwarding Enabled", ifaceName)
+	return invokePSCommand(cmd)
+}
+
+func invokePSCommand(cmd string) error {
+	// Create a backend shell.
+	back := &backend.Local{}
+
+	// start a local powershell process
+	shell, err := ps.New(back)
+	if err != nil {
+		panic(err)
+	}
+	defer shell.Exit()
+	_, stderr, err := shell.Execute(cmd)
+	if err != nil {
+		return err
+	}
+	if stderr != "" {
+		return fmt.Errorf("%s", stderr)
+	}
+	return nil
+}
+
+func callPSCommand(cmd string) (string, error) {
+	// Create a backend shell.
+	back := &backend.Local{}
+
+	// start a local powershell process
+	shell, err := ps.New(back)
+	if err != nil {
+		panic(err)
+	}
+	defer shell.Exit()
+	stdout, stderr, err := shell.Execute(cmd)
+	if err != nil {
+		return stdout, err
+	}
+	if stderr != "" {
+		return stdout, fmt.Errorf("%s", stderr)
+	}
+	return stdout, nil
+}
+
+// RemoveManagementInterface removes the management interface of the HNS Network, and then the physical interface is able
+// to add on the OVS bridge. This function is called only if Hyper-V feature is installed on the host.
+func RemoveManagementInterface(networkName string) error {
+	var err error
+	var maxRetry = 3
+	var i = 0
+	cmd := fmt.Sprintf("Get-VMSwitch -Name %s  | Set-VMSwitch -AllowManagementOS $false ", networkName)
+	// Retry the operation here because an error is returned at the first invocation.
+	for i < maxRetry {
+		err = invokePSCommand(cmd)
+		if err == nil {
+			return nil
+		}
+		i++
+	}
+	return err
+}
+
+// WindowsHyperVInstalled checks if the Hyper-V feature is enabled on the host.
+func WindowsHyperVInstalled() (bool, error) {
+	cmd := "$(Get-WindowsFeature Hyper-V).InstallState"
+	result, err := callPSCommand(cmd)
+	if err != nil {
+		return true, err
+	}
+	return strings.HasPrefix(result, "Installed"), nil
+}
+
+// CreateHNSNetwork creates a new HNS Network, whose type is "Transparent", and NetworkAdapter is parsed from the
+// environment variable "UPLINK_NET_ADAPTER". If the NetworkAdapter is not configured in the environment variables,
```

Comment:
I did not see you read environment var to get the adapter?

Reply:
A reminder on this one.

Reply:
Looks I forgot to save the change, will update.